test/e2e/helpers_test.go

```diff
@@ -528,31 +527,48 @@ func verifyIngressWithEchoBackendsInAdminAPI(
 	t.Helper()
 
 	require.Eventually(t, func() bool {
-		d, err := dump.Get(ctx, kongClient, dump.Config{})
+		start := time.Now()
+		defer func() {
+			t.Logf("Fetched config from %q, started at %s, duration %v", kongClient.BaseRootURL(), start.Format(time.RFC3339), time.Since(start))
+		}()
+
+		services, err := kongClient.Services.ListAll(ctx)
 		if err != nil {
-			t.Logf("failed dumping config: %s", err)
+			t.Logf("failed to list services: %v", err)
 			return false
 		}
-		if len(d.Services) != 1 {
-			t.Log("still no service found...")
+		if len(services) != 1 || services[0].ID == nil {
+			t.Logf("%d services found, expected 1", len(services))
 			return false
 		}
-		if len(d.Routes) != 1 {
-			t.Log("still no route found...")
+
+		routes, _, err := kongClient.Routes.ListForService(ctx, services[0].ID, &kong.ListOpt{Size: 10})
```

Comment:
nit: why do we choose `10` for this?

Reply:
Just an arbitrary value. I think we can leave it empty to use the "default" value.